cpp/src/arrow/adapters/orc/adapter_util.cc

```diff
@@ -315,13 +342,462 @@ Status AppendBatch(const liborc::Type* type, liborc::ColumnVectorBatch* batch,
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::FLOAT:
+      return WriteNumericBatch<arrow::NumericArray<arrow::FloatType>,
+                               liborc::DoubleVectorBatch, double>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::DOUBLE:
+      return WriteNumericBatch<arrow::NumericArray<arrow::DoubleType>,
+                               liborc::DoubleVectorBatch, double>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::BINARY:
+      return WriteBinaryBatch<arrow::BinaryArray, int32_t>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::LARGE_BINARY:
+      return WriteBinaryBatch<arrow::LargeBinaryArray, int64_t>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::STRING:
+      return WriteBinaryBatch<arrow::StringArray, int32_t>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::LARGE_STRING:
+      return WriteBinaryBatch<arrow::LargeStringArray, int64_t>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::FIXED_SIZE_BINARY:
+      return WriteFixedSizeBinaryBatch(column_vector_batch, arrow_offset, orc_offset,
+                                       length, array, incoming_mask);
+    case arrow::Type::type::DATE32:
+      return WriteNumericBatch<arrow::NumericArray<arrow::Date32Type>,
+                               liborc::LongVectorBatch, int64_t>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask);
+    case arrow::Type::type::DATE64:
+      return WriteTimestampBatch<arrow::Date64Array>(
+          column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask,
+          kOneSecondMillis, kOneMilliNanos);
+    case arrow::Type::type::TIMESTAMP: {
+      switch (arrow::internal::checked_pointer_cast<arrow::TimestampType>(array.type())
+                  ->unit()) {
+        case arrow::TimeUnit::type::SECOND:
+          return WriteTimestampBatch<arrow::TimestampArray>(
+              column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask,
+              1, kOneSecondNanos);
+        case arrow::TimeUnit::type::MILLI:
+          return WriteTimestampBatch<arrow::TimestampArray>(
+              column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask,
+              kOneSecondMillis, kOneMilliNanos);
+        case arrow::TimeUnit::type::MICRO:
+          return WriteTimestampBatch<arrow::TimestampArray>(
+              column_vector_batch, arrow_offset, orc_offset, length, array, incoming_mask,
+              kOneSecondMicros, kOneMicroNanos);
+        default:
```

Comment:
it is probably not a good idea of have an implicit default here (you should spell out nanos and default should be returning an error.

Reply:
Thanks! Fixed!